### ISSUE TITLE
<script> tag must not be included in a <div>

### ISSUE THREAD

**Issue by ghost** [holidaycheck/react-google-tag-manager#52]
After paste <GoogleTagManager gtmId="GTM-mycode" /> after body opening i get error: `<script> tag must not be included in a <div>`
How can i fix? Can i remove wrappers of example?

**Comment by FQ400** [comment]
Hey @mlimasolucoes, 

the wrapper is used for a single purpose, returning a single node. See the [documentation](https://facebook.github.io/react/tips/maximum-number-of-jsx-root-nodes.html). As a quickfix you could change the wrapper to an element that doesn't throw an error. You can find a list of them [here](https://support.google.com/tagassistant/answer/3207128).

**Comment by gf3** [comment]
according to the GTM documentation the script tag needs to be a direct child of the body (god know's why—ugh) however that's not really possible with most react apps

**Comment by ghost** [comment]
Was just a warning.
[]'s

**Comment by gf3** [comment]
i fixed it by not including the `<script>` in `render()` and simply doing the `eval` in `componentDidMount()`

**Comment by gf3** [comment]
e.g.:

``` jsx
type DefaultProps = {
  additionalEvents: Object,
  dataLayerName: string,
  scriptId: string
};

type Props = DefaultProps & {
  gtmId: string
};

class GoogleTagManager extends React.Component<DefaultProps, Props, void> {
  static defaultProps: DefaultProps = {
    additionalEvents: {},
    dataLayerName: 'dataLayer',
    scriptId: 'react-google-tag-manager-gtm'
  };

  componentDidMount() {
    const gtm = buildGTM({
      id: this.props.gtmId,
      additionalEvents: this.props.additionalEvents,
      dataLayerName: this.props.dataLayerName
    });

    if (!window[this.props.dataLayerName]) {
      eval(gtm.scriptAsHTML().replace(/<\/?script>/g, ''));
    }
  }

  render() {
    const gtm = buildGTM({
      id: this.props.gtmId,
      additionalEvents: this.props.additionalEvents,
      dataLayerName: this.props.dataLayerName
    });

    return gtm.noScriptAsReact();
  }
}

export default GoogleTagManager;
```

**Comment by lucafik** [comment]
can someone confirm if the GTM also works if inside a container, as in the example?

Or is there some kind of container that is recommended, because according to google's documentation it should be completely avoided to nest it.

**Comment by shri3k** [comment]
I think it's obvious for most but it's worth mentioning that if you're doing server-side rendering then you can skip the `componentDidMount` hook altogether.

**Comment by gf3** [comment]
@shri3k it wouldn't be isomorphic then

**Comment by shri3k** [comment]
Unless for some reason your server side fails to load (on which I think you have bigger probelm than not loading this component) then yes, you'll need to rely on `componentDidMount` to load this for you. But I've always felt that isomorphic meant server side first and then hand off to client. Never the other way around.

**Comment by pacarvalho** [comment]
Is there an official fix for this issue? 

![image](https://user-images.githubusercontent.com/860599/42711962-36d4d854-86c0-11e8-818c-952e9b8806d3.png)

Is this even a problem? Or is this error being detected because at the moment when the page first loads react hasn't replaced the DOM?
